File: Recomendação_Dia_5/usuario.py
from recomendacao import recomendacao
import requests
import pandas as pd

print('Escolha o usuario digitando o seu id:')
valor = input()

try:
    text = requests.get(f'http://localhost:5163/usuario/{valor}')

    usuario = pd.read_json(text.text, encoding='UTF-8')

    avaliacao = pd.json_normalize(usuario['avaliacao'])

    usuario = usuario.join(avaliacao, how='left')
    usuario.drop('avaliacao', axis=1, inplace=True)
    usuario = usuario[['id', 'filmeId', 'nota', 'idade', 'genero']]
    usuario['genero'] = usuario['genero'].replace({'M': 1, 'F': 0})

    colunas = ['user id', 'item id', 'rating', 'age', 'gender']
    usuario.columns = colunas

    filmes = recomendacao(usuario)
    filmes = filmes['movie title'].values

    print('\nRecomendação de filmes:')
    print(f'1- {filmes[0]}')
    print(f'2- {filmes[1]}')
    print(f'3- {filmes[2]}')
    print(f'4- {filmes[3]}')
    print(f'5- {filmes[4]}\n')

except:
    print('Usuario não existe, programa encerrado.')

File: Recomendação_Dia_5/ModeloKmeans.API/Models/Usuario.cs
namespace ModeloKmeans.Models;

internal class Usuario
{
    private int id;
    private int idade;
    private string genero;
    private IEnumerable<Avaliacao> avaliacao;

    public int Id
    {
        get
        {
            return id;
        }

        private set
        {
            id = value;
        }
    }
    
    public int Idade
    {

        get
        {
            return idade;
        }

        set
        {
            idade = value;
        }
    }
    
    public string Genero
    {

        get
        {
            return genero;
        }

        set
        {
            genero = value;
        }
    }
    
    public IEnumerable<Avaliacao> Avaliacao
    {

        get
        {
            return avaliacao;
        }

        set
        {
            avaliacao = value;
        }
    }
    

    public Usuario(int Id, IEnumerable<Avaliacao> Avaliacao, int Idade, string Genero)
    {
        this.Id = Id;
        this.Avaliacao = Avaliacao;
        this.Idade = Idade;
        this.Genero = Genero;
    }
}